internal/vault/cache.go

```diff
@@ -134,7 +157,6 @@ func (c *clientCache) Prune(filterFunc ClientCachePruneFilterFunc) []ClientCache
 }
 
 func (c *clientCache) remove(key ClientCacheKey, client Client) bool {
-	client.Close()
```

Comment:
I don't recall why this was changed, why is `client.Close()` removed here?

Reply:
[Our onEvict function calls `client.Close()`](https://github.com/hashicorp/vault-secrets-operator/blob/main/internal/vault/client_factory.go#L123C37-L134), and the lru cache remove function calls this onEvict function `github.com/hashicorp/golang-lru@v0.5.4/simplelru/lru.go:175`.